modules/system/assets/ui/js/filter.js

```diff
@@ -295,13 +295,20 @@
             item = $.grep(fromItems, testFunc).pop(),
             filtered = $.grep(fromItems, testFunc, true)
 
-        if (isDeselect)
+        if (isDeselect) {
             this.scopeValues[this.activeScopeName] = filtered
-        else
+        } else {
             this.scopeAvailable[this.activeScopeName] = filtered
+        }
 
-        if (item)
+        if (item) {
             toItems.push(item)
+        } else {
+            toItems.push({
+                'id': itemId,
+                'name': $item.text()
+            })
```

Comment:
Why was this change required? Also, what is this variable even used for?

Reply:
@LukeTowers 

As the searched items added only to the DOM (group filter's `<ul>`), we need to add the item to `scopeValues` or `scopeAvailable` array when it is clicked.

If `item` is selected `toItems` equals `scopeValues`. If `item` is deselected, `toItems` equals `ScopeAvailable`. 

The `selectItem` method checks the clicked `item` object existance in `scopeValues` and in `scopeAvailable` and push it to the opposite (according it's selected or deselected). If the `item` doesn't exist in none of the arrays, we create the object from the clicked `<li>` and push it anyway.

Reply:
I'm not seeing where the `toItems` variable is used though, could you link me to it?

Reply:
@danielbidala ^

Reply:
@LukeTowers I'm not sure I understand your question. The `toItems` array is just a reference to `scopeValues` or `scopeAvailable` array depending on whether the element was selected or deselected. It is used only in `selectItem` method

Reply:
@bennothommo maybe you know what I'm asking? It looks like the toItems variable is populated here but never actually used anywhere else. I don't see a reason for it to exist period.

Reply:
@LukeTowers yeah, it appears to be unused. I looked through the Git blame and saw it was added about 7 years ago in its original form, but wasn't even used then as far as I can tell.

Reply:
Let's just get rid of it then. @danielbidala not sure why a change was needed to a variable that doesn't get used, can you explain the logic behind that?

Reply:
@LukeTowers The logic was there I've just extended it :) Anyway I added a commit to remove toItems variable.